packages/components/src/unit-control/index.tsx

```diff
@@ -76,10 +72,18 @@ function UnforwardedUnitControl(
 	// ensures it fallback to `undefined` in case a consumer of `UnitControl`
 	// still passes `null` as a `value`.
 	const nonNullValueProp = valueProp ?? undefined;
-	const units = useMemo(
-		() => getUnitsWithCurrentUnit( nonNullValueProp, unitProp, unitsProp ),
-		[ nonNullValueProp, unitProp, unitsProp ]
-	);
+	const [ units, reFirstCharacterOfUnits ] = useMemo( () => {
+		const list = getUnitsWithCurrentUnit(
+			nonNullValueProp,
+			unitProp,
+			unitsProp
+		);
+		const firstCharacters = list.reduce( ( carry, { value } ) => {
+			const first = value.substr( 0, 1 );
+			return carry.includes( first ) ? carry : `${ carry }|${ first }`;
+		}, list[ 0 ]?.value.substr( 0, 1 ) );
+		return [ list, new RegExp( `^(?:${ firstCharacters })$` ) ];
```

Comment:
This might be a bit of a contrived edge case but should we make this regex case insensitive? 

Without that, I can get an `E` to be entered into the input using either capslock or holding shift. It also means if capslock is on for some reason the unit change doesn't work.

<img width="283" alt="Screen Shot 2022-03-23 at 11 22 41 am" src="https://user-images.githubusercontent.com/60436221/159603701-a8cab05c-7436-4238-9154-36b674aac1bd.png">

```suggestion
		return [ list, new RegExp( `^(?:${ firstCharacters })$`, 'i' ) ];
```

Reply:
This is a great point. I think case-insensitive makes sense.

One thing I also wanted to consider was moving the focus when any character besides a numeral is input. But the most obvious win for that may just be simpler code and I'm not sure that it helps UX.